feat(d2k): add AI Battle spectator mode

YibinLong opened the pull request:
Implement a new game mode where players can watch AI factions battle each other.

Features:
- AI Battle state management and game integration
- Setup screen for configuring AI factions, map, and game settings
- Observer UI for watching battles with timeline scrubber
- Results screen showing battle outcomes
- Replay functionality for reviewing completed battles
- Integration with D2K mod main menu

YibinLong committed:
feat(d2k): add AI Battle spectator mode

Implement a new game mode where players can watch AI factions battle each other.

Features:
- AI Battle state management and game integration
- Setup screen for configuring AI factions, map, and game settings
- Observer UI for watching battles with timeline scrubber
- Results screen showing battle outcomes
- Replay functionality for reviewing completed battles
- Integration with D2K mod main menu

PunkPun reviewed (commented):
We expect all PR's to pass the lint tests, as described in https://github.com/OpenRA/OpenRA/wiki/Contributing

PunkPun reviewed (commented):
This seems like a really cool feature, however it is mostly duplicating already existing logic. It may be better to streamline the existing skirmish UI rather than adding a less-featureful duplicate. And just to integrate other features into the rest of the game better.

I know there was a request for something like this in #21461, however we didn't flesh out the ideas.

Changing the simulation speed ingame might cause all sorts of issues as some calculations are cached with the idea that speed never changes. Would need to screen for regressions.

However adding a simulation speed option to regular gameplay would be very cool. Total War'esque. The pause would become exposed to UI. Just in multiplayer the speed controls would be off. Maybe it could be implemented by adding a separate option to game speed controlls in lobby option dropdown: "Controllable". 

It would be cool to add end-game results to the end of every match like all the CnC games. It has been a long desired feature. However it should be done without exiting the match. #8775 #11568

It might also be a good idea to enable debug commands in such an AI spectator scenario?

Also it doesn't appear like the rewind functionality works? It just resets the replay. However if it did work correctly, adding such replay controls to observer UI would be incredibly useful. #14104 #15181 #18121

YibinLong committed:
fix: improve AI battle UI consistency and localization

Standardized number formatting using CultureInfo.InvariantCulture in AI battle logic and results. Refactored code for clarity and maintainability, including use of FluentReference for replay labels and improved order construction. Updated D2K mod UI and localization files to add missing slot header label and clean up unused tooltips.

YibinLong commented:
Hey, thank you for the review! Indeed this PR was based off of #21461, I missed adding it to my description.

That is a good idea to add the speedup to the skirmish UI so we avoid duplicating features. I'm also thinking of moving the end game results to the end of every match, but yeah that'll just require a bit more tinkering if we want the CnC style end-game results. Also +1 on the debug commands.

For the rewind, yea, I'm thinking of just scrapping that feature for now since it's not working too well, and make it a whole separate PR if someone wants to tackle rewinding to certain times, because it's going to require quite a bit more work to get that working nicely.

YibinLong committed:
feat: add Claude agent and command documentation files

Introduces new markdown documentation for Claude agents (codebase-analyzer, codebase-locator, codebase-pattern-finder, web-search-researcher, thoughts-analyzer, thoughts-locator) and commands (commit, create_plan, describe_pr, implement_plan, iterate_plan, research_codebase) in the .claude directory. Also adds coding standards, linter fix notes, technical documentation, and implementation plans and research documents for D2K tutorial and AI battle features. Updates mainmenu.yaml in mods/d2k/chrome.
